File: backend/src/main/java/vn/fpt/se18/MentorLinking_BackEnd/service/admin/ReviewManagementService.java
package vn.fpt.se18.MentorLinking_BackEnd.service.admin;

import vn.fpt.se18.MentorLinking_BackEnd.dto.request.BaseRequest;
import vn.fpt.se18.MentorLinking_BackEnd.dto.request.admin.review.ReviewFilterRequest;
import vn.fpt.se18.MentorLinking_BackEnd.dto.response.BaseResponse;
import vn.fpt.se18.MentorLinking_BackEnd.dto.response.PageResponse;
import vn.fpt.se18.MentorLinking_BackEnd.dto.response.admin.ReviewDetailResponse;
import vn.fpt.se18.MentorLinking_BackEnd.dto.response.admin.ReviewStatisticsResponse;

public interface ReviewManagementService {
    
    BaseResponse<PageResponse<ReviewDetailResponse>> getAllReviews(BaseRequest<ReviewFilterRequest> request);
    
    BaseResponse<ReviewDetailResponse> getReviewById(Long id);
    
    BaseResponse<Void> publishReview(Long id);
    
    BaseResponse<Void> unpublishReview(Long id);
    
    BaseResponse<Void> deleteReview(Long id);
    
    BaseResponse<Void> updateModerationNote(Long id, String note);
    
    BaseResponse<Void> bulkPublish(Long[] ids);
    
    BaseResponse<Void> bulkDelete(Long[] ids);
    
    BaseResponse<ReviewStatisticsResponse> getStatistics();
}

File: frontend/src/services/admin/reviewManagementService.js
import { instance as axios } from '../../api/axios';

/**
 * Admin Review Management Service
 * Handles reviews/ratings management from bookings
 */

// Get all reviews with filtering and pagination
export const getAllReviews = async (filters) => {
    const response = await axios.post('/api/admin/reviews/list', {
        data: {
            keySearch: filters.keySearch || null,
            rating: filters.rating || null,
            status: filters.status || null,
            dateFrom: filters.dateFrom || null,
            dateTo: filters.dateTo || null,
            page: filters.page || 1,
            size: filters.size || 10
        }
    });
    return response;
};

// Get review details by ID
export const getReviewById = async (id) => {
    const response = await axios.get(`/api/admin/reviews/${id}`);
    return response;
};

// Publish review
export const publishReview = async (id) => {
    const response = await axios.put(`/api/admin/reviews/${id}/publish`);
    return response;
};

// Unpublish/Hide review
export const unpublishReview = async (id) => {
    const response = await axios.put(`/api/admin/reviews/${id}/unpublish`);
    return response;
};

// Delete review
export const deleteReview = async (id) => {
    const response = await axios.delete(`/api/admin/reviews/${id}`);
    return response;
};

// Update moderation note
export const updateModerationNote = async (id, note) => {
    const response = await axios.put(`/api/admin/reviews/${id}/moderation-note`, null, {
        params: { note }
    });
    return response;
};

// Bulk publish reviews
export const bulkPublishReviews = async (ids) => {
    const response = await axios.put('/api/admin/reviews/bulk-publish', ids);
    return response;
};

// Bulk delete reviews
export const bulkDeleteReviews = async (ids) => {
    const response = await axios.delete('/api/admin/reviews/bulk-delete', {
        data: ids
    });
    return response;
};

// Get review statistics
export const getReviewStatistics = async () => {
    const response = await axios.get('/api/admin/reviews/statistics');
    return response;
};

export default {
    getAllReviews,
    getReviewById,
    publishReview,
    unpublishReview,
    deleteReview,
    updateModerationNote,
    bulkPublishReviews,
    bulkDeleteReviews,
    getReviewStatistics
};
